orangecontrib/prototypes/widgets/owexplpred.py

```diff
@@ -0,0 +1,401 @@
+        self.Warning.selection_not_matching.clear()
+        if self.data is not None and self.toExplain is not None:
+            if domain_equal(self.data.domain, self.toExplain.domain):
+                if self.model is not None:
+                    # calculate contributions
+                    e = ExplainPredictions(self.data,
+                                           self.model,
+                                           batchSize=min(
+                                               int(len(self.data.X) / 5), 100),
+                                           pError=self.gui_p_val / 100,
+                                           error=self.gui_error / 100)
+                    self._task = task = Task()
+
+                    def callback():
+                        nonlocal task
+                        if task.canceled:
+                            return True
+                        return False
+                    explain_func = partial(
+                        e.anytime_explain, self.toExplain, callback=callback)
+                    task.future = self._executor.submit(explain_func)
+                    task.watcher = FutureWatcher(task.future)
+                    task.watcher.done.connect(self._task_finished)
+            else:
+                self.Warning.selection_not_matching()
+        else:
+            self.Outputs.explanations.send(None)
+
+    @pyqtSlot(concurrent.futures.Future)
+    def _task_finished(self, f):
+        """
+        Parameters:
+        ----------
+        f: conncurent.futures.Future
+            future instance holding the result of learner evaluation
+        """
+        assert self.thread() is QThread.currentThread()
+        assert self._task is not None
+        assert self._task.future is f
+        assert f.done()
+
+        self._task = None
+
+        try:
+            results = f.result()
+        except Exception as ex:
+            log = logging.getLogger()
+            log.exception(__name__, exc_info=True)
+            self.error("Exception occured during evaluation: {!r}".format(ex))
+
+            for key in self.results.keys():
+                self.results[key] = None
+        else:
+            classValue = results[0]
+            self.explanations = results[1]
+            self._print_prediction(classValue)
+            self.Outputs.explanations.send(self.explanations)
+            model = TableModel(self.explanations, parent=None)
+            self.dataview.setModel(model)
+
+    def cancel(self):
+        """
+        Cancel the current task (if any).
+        """
+        if self._task is not None:
+            self._task.cancel()
+            assert self._task.future.done()
+            # disconnect the `_task_finished` slot
+            self._task.watcher.done.disconnect(self._task_finished)
+            self._task_finished(self._task.future)
+
+    def _print_prediction(self, classValue):
+        """
+        Parameters
+        ----------
+        classValue: float 
```

Comment:
https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

(and all occurances of this variable, and maybe some other ones too...)